cli/src/commands/walletconnect.js

```diff
@@ -0,0 +1,286 @@
+      const unsub = listen((l: any) => {
+        if (opts.verbose) {
+          o.next(l.type + ": " + l.message);
+        } else if (!opts.silent && l.type === "walletconnect") {
+          o.next(l.message);
+        }
+      });
+
+      let account = await scan(opts).pipe(take(1)).toPromise();
+
+      if (!account) {
+        throw new Error("No account");
+      }
+
+      log("walletconnect", account);
+
+      const connector = new WalletConnect(
+        opts.walletConnectSession
+          ? {
+              session: JSON.parse(opts.walletConnectSession),
+            }
+          : {
+              // Required
+              uri: opts.walletConnectURI,
+              // Required
+              clientMeta: {
+                description: "LedgerLive CLI",
+                url: "https://ledger.fr",
+                icons: [
+                  "https://avatars0.githubusercontent.com/u/9784193?s=400&v=4",
+                ],
+                name: "LedgerLive CLI",
+              },
+            }
+      );
+
+      connector.on("session_request", (error, payload) => {
+        if (error) {
+          throw error;
+        }
+
+        log("walletconnect", "session_request");
+        log("walletconnect", payload);
+
+        connector.approveSession({
+          accounts: [account.freshAddress],
+          chainId: account.currency.ethereumLikeInfo.chainId,
+        });
+      });
+
+      connector.on("call_request", async (error, payload) => {
+        if (error) {
+          throw error;
+        }
+
+        log("walletconnect", "call_request");
+        log("walletconnect", payload);
+
+        let transactionRaw;
+        let message;
+        let errorMsg;
+        let result;
+
+        try {
+          switch (payload.method) {
+            case "eth_signTypedData":
+            case "eth_sign":
+              payload.params = payload.params.reverse();
+            case "personal_sign":
+              message = {
+                path: account.freshAddressPath,
+                message:
+                  payload.method === "eth_signTypedData"
+                    ? JSON.parse(payload.params[0])
+                    : Buffer.from(payload.params[0].slice(2), "hex").toString(),
+                currency: getCryptoCurrencyById("ethereum"),
+                derivationMode: account.derivationMode,
+              };
+
+              log("walletconnect", "message to sign");
+              log("walletconnect", (message: any));
+
+              result = await withDevice(opts.device || "")((t) =>
+                from(signMessage(t, message))
+              )
+                .pipe(take(1))
+                .toPromise();
+              result = result.signature;
+
+              log("walletconnect", "message signature");
+              log("walletconnect", result);
+
+              break;
+            case "eth_signTransaction":
+            case "eth_sendTransaction":
+              transactionRaw = payload.params[0];
```

Comment:
wording choice: probably use another name than "transactionRaw" to not confuse with the `TransactionRaw` type in live-common.